### PR TITLE
ci: enforce version of Maven

### PR DESCRIPTION
**Pull request by triceo**
`MAVEN_ARGS` is only supported since 3.9.0.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -48,6 +48,11 @@ jobs:
           distribution: 'temurin'
           cache: 'maven'
 
+      - name: Set up Maven
+        uses: stCarolas/setup-maven@v4.5
+        with:
+          maven-version: 3.9.3
```
Comment (rsynek) [inline .github/workflows/release.yml:54]: Is the maintenance cost (upgrading the maven version over all the release yaml files) worth the MAVEN_ARGS support?
Alternative: use just any env var ( that does not clash with any other ) and use its value in the maven command.

Comment (triceo) [reply .github/workflows/release.yml:54]: I would argue that not being victims of Github's arbitrary version update policy is the main benefit here. *We* decide when Maven changes, not Github.

Comment (triceo) [reply .github/workflows/release.yml:54]: (The current Maven in GHA is 3.8.8.)

Comment (rsynek) [reply .github/workflows/release.yml:54]: I don't care so much until is works. If something fails due to a silent upgrade, it's a different story. But I get it, let's merge this.

**Review by rsynek — APPROVED** [review]

**Pull request merged**